pool.go

```diff
@@ -363,11 +373,13 @@ func (p *Pool[T]) acquire(ctx context.Context) (*Resource[T], error) {
 
 	// If a resource is available in the pool.
 	if res := p.tryAcquireIdleResource(); res != nil {
+		waitTime := time.Duration(nanotime() - startNano)
```

Comment:
Not directly related to your PR as you are reusing something that existed, but I'm curious

Any reason to use `startNano := nanotime`, and not a `start := time.Now()` and here a `time.Now().Since(start)`?

Reply:
IDK, as you said I am reusing previous approach to get time. `nanotime` is provided via linking so I assume there is some platform specific trickery to make it cheaper to call maybe?

Reply:
Yes, directly calling `nanotime` is a micro optimization. 

On my machine an uncontested Acquire and Release cycle takes 103ns with time.Now but only 56ns with nanotime directly.

Reply:
OK. So maybe a comment about it could be interesting, because the pattern is not obvious.

Or maybe I missed it